### PR TITLE
Fix upgrade check (again)

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix upgrade check

3 changes:
* Upgrade stateless settings ref to 24.7.
* Move `merge_tree_min_bytes_per_task_for_remote_reading` to 24.8.
* Change format of SettingsChangesHistory.cpp to try to make it easier to edit and merge conflicts automatically.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit baae45889bad6bb6827278ca677025b01dab1da7 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67524/baae45889bad6bb6827278ca677025b01dab1da7/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67524/baae45889bad6bb6827278ca677025b01dab1da7/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by Avogar — APPROVED** [review]

File: src/Core/SettingsChangesHistory.cpp [hunk 1]
```diff
@@ -57,6 +57,446 @@ String ClickHouseVersion::toString() const
...
+    {"19.5",
+        {
+            {"max_partitions_per_insert_block", 0, 100, "Add a limit for the number of partitions in one block"}
+        }
+    },
+    {"18.12.17",
+        {
+            {"enable_optimize_predicate_expression", 0, 1, "Optimize predicates to subqueries by default"}
+        }
+    },
```
Comment (antaljanosbenjamin) [inline src/Core/SettingsChangesHistory.cpp:499]: Tihs doesn't look good, seems like everything is duplicated.

Comment (Avogar) [reply src/Core/SettingsChangesHistory.cpp:499]: Seems like last merge with master went wrong

Comment (Algunenano) [reply src/Core/SettingsChangesHistory.cpp:499]: Yep, my bad. We have `merge=union` setup and it merged cleanly locally, which was odd, so I looked into the changes and applied them in my file, without noticing the old ones were kept too. Fixing it

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**